support-workers/src/test/scala/com/gu/support/workers/errors/ErrorHandlerSpec.scala

```diff
@@ -93,10 +93,15 @@ class ErrorHandlerSpec extends AnyFlatSpec with Matchers {
     ).asRetryException shouldBe a[
       RetryUnlimited,
     ]
+    ZuoraErrorResponse(
+      false,
+      List(ZuoraError("TEMPORARY_ERROR", "Operation failed due to a temporary error, please retry later.")),
+    ).asRetryException shouldBe a[
+      RetryUnlimited,
```

Comment:
Went for unlimited rather than 5-times? Guessing its unlikely the a server would be back up 5 retries

Reply:
My logic is that it is temporary - so should come back up eventually.

I'm not to sure what the usecase for `RetryLimited` is, potentially if a service is overwhelmed and we want to backoff to allow it to stabilise. 

[We also use it for unknown errors](https://github.com/guardian/support-frontend/blob/main/support-workers/src/main/scala/com/gu/support/workers/exceptions/ErrorHandler.scala#L66-L67) - you can see the other usecases in this file.

Reply:
@jamesgorrie  unlimited is 24h, limited is about 2 hours I think (exponential backoff)
The main issue with unlimited is that the state machine times out completly after 24h and the customer doesn't get an email afaik so they don't realise it's failed.  Also we don't get a "failed" alarm until 24h later.  The limited one they get an email telling them it's failed, so they can take action e.g. retry.

Reply:
Thanks @johnduffell - I think in this instance then we probably want to `RetryLimited` - if Zuora's `TEMPORARY_ERROR` is more than 2 hours, I imagine it's worth knowing about then.

The example that inspired this PR it seems like it was out for < 5 minutes. So that fits the bill too.

[Here's the PR](https://github.com/guardian/support-frontend/pull/5947)